runtime/src/main/java/org/corfudb/util/GitRepositoryState.java

```diff
@@ -76,4 +76,19 @@ public static GitRepositoryState getRepositoryState() {
         return _gitRepositoryState;
     }
 
+    /**
+     * Getter for the long value which will be populated into our protobuf header. We only
+     * take the first CUTOFF characters of parent git commit id as they are more than enough
+     * to uniquely identify a commit in our codebase.
+     *
+     * @return Long value converted from the first 12 characters of parent git commit id.
+     */
+    public static long getCorfuSourceCodeVersion() {
+        if (_gitRepositoryState == null) {
+            getRepositoryState();
+        }
+        final int CUTOFF = 12;
+        final int HEX_OPT = 16;
+        return Long.parseLong(_gitRepositoryState.commitId.substring(0, CUTOFF), HEX_OPT);
+    }
```

Comment:
This has a race condition. Writing to `_gitRepositoryState` and accessing it can happen from multiple threads but it's not synchronized. 

`_gitRepositoryState` can be made a singleton.

Reply:
Revised, thanks a lot for pointing out this!